docs/cookbook/recipe_customizing_a_mosaic_list.rst

```diff
@@ -98,6 +99,65 @@ for instance the MediaBundle defines the method as::
         }
     }
 
+.. note::
+    In your own admin, media is just a field and not the ``$object``. Therefore,
+    the code above must be updated this way:
+
+    .. code-block:: php
+
+        <?php
+
+            // ...
+
+            public function getObjectMetadata($object)
+            {
+                $media = $object->getMediaField();
+
+                $provider = $this->pool->getProvider($media->getProviderName());
+
+                $url = $provider->generatePublicUrl($media, $provider->getFormatName($media, 'admin'));
+
+                return new Metadata($media->getName(), $media->getDescription(), $url);
+            }
+
+    You will also have to use dependency injection. For this, first define
+    the ``$pool`` variable and override the constructor:
+
+    .. code-block:: php
+
+        <?php
+
+            // ...
+
+            private $pool;
+
+            // ...
+
+            public function __construct(string $code, string $class, string $baseControllerName, Pool $pool)
+            {
+                parent::__construct($code, $class, $baseControllerName);
+
+                $this->pool = $pool;
```

Comment:
its a personal opinion but I would assign the variable first and call the parent one afterwards

WDYT @greg0ire @kunicmarko20 @core23 ?

Reply:
Isn't that exactly what is being done here?

Reply:
Yes now it is done 😊